apps/prairielearn/elements/pl-string-input/pl-string-input.py

```diff
@@ -70,47 +86,42 @@ def render(element_html: str, data: pl.QuestionData) -> str:
     name = pl.get_string_attrib(element, "answers-name")
     label = pl.get_string_attrib(element, "label", LABEL_DEFAULT)
     suffix = pl.get_string_attrib(element, "suffix", SUFFIX_DEFAULT)
-    display = pl.get_enum_attrib(element, "display", DisplayType, DISPLAY_DEFAULT)
-    remove_leading_trailing = pl.get_boolean_attrib(
-        element, "remove-leading-trailing", REMOVE_LEADING_TRAILING_DEFAULT
-    )
+
     remove_spaces = pl.get_boolean_attrib(
         element, "remove-spaces", REMOVE_SPACES_DEFAULT
     )
     placeholder = pl.get_string_attrib(element, "placeholder", PLACEHOLDER_DEFAULT)
     show_score = pl.get_boolean_attrib(element, "show-score", SHOW_SCORE_DEFAULT)
 
     raw_submitted_answer = data["raw_submitted_answers"].get(name)
-
+    multiline = pl.get_boolean_attrib(element, "multiline", MULTILINE_DEFAULT)
     score = data["partial_scores"].get(name, {"score": None}).get("score", None)
     parse_error = data["format_errors"].get(name)
 
+    # Defaults here depend on multiline
+    display = pl.get_enum_attrib(
+        element,
+        "display",
+        DisplayType,
+        DisplayType.BLOCK if multiline else DISPLAY_DEFAULT,
+    )
+
+    if display is DisplayType.INLINE and multiline:
+        raise ValueError('Display must be set to "block" for multiline input.')
```

Comment:
Tweaking this slightly so that we avoid giving the impression that one must explicitly set `display="block"` in this case.

```suggestion
        raise ValueError('Display cannot be "inline" for multiline input.')
```